src/services/test/validate-event.test.ts

```diff
@@ -103,9 +130,173 @@ describe('event-validation', () => {
+      component_id: "TICF_CRI",
+      user: {user_id: "urn:fdc:gov.uk:2022:USER_ONE"},
+      extensions: {
+        intervention: {
+          intervention_code: "01",
+          intervention_reason: "something",
+          originating_component_id: "CMS",
+          originator_reference_id: "1234567",
+          requester_id: "1234567"
+        }
+      }
+    };
+
+    await expect(async () =>
+      await validateEventIsNotStale(EventsEnum.FRAUD_SUSPEND_ACCOUNT, staleEvent, dynamoDBResult)
+    ).rejects.toThrow(new ValidationError('Event received predates last applied event for this user.'));
+    expect(logAndPublishMetric).toHaveBeenCalledWith(MetricNames.INTERVENTION_EVENT_STALE);
+    expect(sendAuditEvent).toHaveBeenCalledWith("AIS_INTERVENTION_IGNORED_STALE", "urn:fdc:gov.uk:2022:USER_ONE", {"appliedAt": undefined, "intervention": "FRAUD_SUSPEND_ACCOUNT", "reason": "Received intervention predates latest applied intervention"})
+  });
+
+  it('should not throw if event is not stale', async () => {
+    const nonStaleEvent = {
+      timestamp: timestamp.seconds,
+      event_name: "TICF_ACCOUNT_INTERVENTION",
+      component_id: "TICF_CRI",
+      user: {user_id: "urn:fdc:gov.uk:2022:USER_ONE"},
+      extensions: {
+        intervention: {
+          intervention_code: "01",
+          intervention_reason: "something",
+          originating_component_id: "CMS",
+          originator_reference_id: "1234567",
+          requester_id: "1234567"
+        }
+      }
+    };
+
+    await expect(validateEventIsNotStale(EventsEnum.FRAUD_SUSPEND_ACCOUNT, nonStaleEvent, dynamoDBResult)).resolves.toEqual(undefined);
+    expect(logAndPublishMetric).not.toHaveBeenCalled();
+    expect(sendAuditEvent).not.toHaveBeenCalled();
+  });
+
+  it('should throw an error if event timestamp is in the future', async () => {
+    const eventInTheFuture = {
+      timestamp: timestamp.seconds + 10,
+      event_name: "TICF_ACCOUNT_INTERVENTION",
+      component_id: "TICF_CRI",
+      user: {user_id: "urn:fdc:gov.uk:2022:USER_ONE"},
+      extensions: {
+        intervention: {
+          intervention_code: "01",
+          intervention_reason: "something",
+          originating_component_id: "CMS",
+          originator_reference_id: "1234567",
+          requester_id: "1234567"
+        }
+      }
+    };
+    await expect(async () => { await validateEventIsNotInFuture(EventsEnum.FRAUD_SUSPEND_ACCOUNT, eventInTheFuture)}).rejects.toThrow(new Error("Event is in the future. It will be retried"));
+    expect(logAndPublishMetric).toHaveBeenCalledWith(MetricNames.INTERVENTION_IGNORED_IN_FUTURE);
+    expect(sendAuditEvent).toHaveBeenCalledWith("AIS_INTERVENTION_IGNORED_IN_FUTURE", "urn:fdc:gov.uk:2022:USER_ONE", {"appliedAt": undefined, "intervention": "FRAUD_SUSPEND_ACCOUNT", "reason": "received event is in the future"});
+  });
+
+  it('should throw not throw an error if event is not in the future', async () => {
```

Comment:
I'm not sure what this test description means, sorry could this be reworded to be clearer?

Reply:
ahah no, good spot. The sentence itself doesn't make sense 😂